APC-SoCIT/APC-2024-2025-T1-05-Admission-Management-System
# Add document upload functionality with validation and storage

# SCRUM-10: Efficient Handling of New Applications and Documents for Admission Officers

## Description
This PR implements the core functionality for admission officers to efficiently manage and process applications and documents seamlessly. The implementation follows the ERD and wireframes from our documentation, providing a complete flow for application management.

## Features Implemented

### Core Authentication & Authorization
- Created admission officers table and model with proper relations
- Added role-based middleware for secure access control
- Implemented role column in users table
- Fixed role verification and middleware registration

### Application Management Interface
- Application listing with status indicators and document count
- Detailed application view with document management
- Status update functionality with history tracking
- Document upload system with validation and storage
- Document viewer with size display and date information

### Admission Officer Dashboard
- Statistics showing pending/approved/rejected applications
- Recent activity tracking system
- Visual status indicators and timeline
- Integrated React components for dynamic updates

## Database Changes
- Added admission_officers table following ERD specifications
- Added role column to users table
- Created application_histories table for activity tracking
- Enhanced documents table with proper relations

## Testing Steps
1. Log in with admission officer credentials
2. Navigate to /applications to view application list
3. Test document upload functionality:
   - Verify file type validation
   - Check size limits
   - Confirm proper storage
4. Test status updates:
   - Update application status
   - Verify history tracking
   - Check notifications

## Related Issues
- Fixes #23 (Admission Officer Application Management)
- Implements SCRUM-10 requirements

## Screenshots
![dc274f07-977e-4425-ae3a-03461de2f76a](https://github.com/user-attachments/assets/cac2df8e-3950-45d7-92b5-eabfd222fe45)
![462558355_1098964232024585_7511466315621471486_n](https://github.com/user-attachments/assets/b26f3fc4-0074-4ef0-a5be-69e114768dd4)
![462636491_1069539591895118_377884467990735345_n](https://github.com/user-attachments/assets/07115ba9-ea76-4af3-9fbc-65c53e717c90)


## Checklist
- [x] Follows ERD structure
- [x] Matches wireframe designs
- [x] Includes proper role verification
- [x] Implements document management
- [x] Adds activity logging
- [x] Includes error handling
- [x] Uses proper middleware
